vars/customWARPackager.groovy

```diff
@@ -0,0 +1,117 @@
+/**
+ * Builds a custom Jenkins WAR using <a href="https://github.com/jenkinsci/custom-war-packager">Custom WAR Packager</a>.
+ * The method supports both YAML BOM and Custom WAR Packager YAM as an input.
+ *
+ * @param metadataFile Path to the metadata file.
+ * @param outputWAR Path, to which the output WAR file should be written
+ * @param outputBOM Path, to which the output BOM file should be written.
+ *                  More info about BOMs: https://github.com/jenkinsci/jep/pull/92
+ * @param mvnSettingsFile
+ *                  Maven Settings file to be used during the build.
+ *                  If {@code null}, it will be determined automatically.
+ * @return nothing
+ */
+def build(String metadataFile, String outputWAR, String outputBOM, String mvnSettingsFile) {
+    def metadata = readYaml(file: metadataFile)
+    def bomFilePath = metadata.packaging.bom ?: null
+    def environment = metadata.packaging.environment
+    def jdk = metadata.packaging.jdk ?: "8"
+    //TODO: replace by a stable release once ready, better name?
+    def cwpVersion = metadata.packaging.cwpVersion ?: "0.1-alpha-5-20180503.142207-7"
+
+    // Resolve the CWP configuration file
+    def configFilePath = null
+    if (metadata.packaging.config != null) {
+        configFilePath = "${pwd tmp: true}/packager-config.yml"
+        sh "rm -rf ${configFilePath}"
+        writeYaml(file: configFilePath, data: metadata.packaging.config)
+    } else if (metadata.packaging.configFile) {
+        configFilePath = metadata.packaging.configFile
+    } else {
+        error "packaging.config or packaging.configFile must be defined"
+    }
+
+    // Resolve the Maven configuration file if not passed
+    if (mvnSettingsFile == null) {
+        mvnSettingsFile = "${pwd tmp: true}/settings-azure.xml"
```

Comment:
Probably I am misunderstanding this, but who is putting this file settings file here?

Reply:
Fixed